### ISSUE TITLE
Task's `failed_when` with `smth in [...]` fails because of templating error

### ISSUE THREAD

**Issue by grossws** [ansible/ansible#15981]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.0.2.0
  config file = /home/gross/experiments/ansible-bug/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

System `/etc/ansible/ansible.cfg` isn't changed.

`/home/gross/experiments/ansible-bug/ansible.cfg`:

``` ini
[defaults]
hostfile = servers
```

`servers`:

``` ini
localhost ansible_python_interpreter=/usr/bin/python2
```
##### OS / ENVIRONMENT

ArchLinux x86_64, python 2.7.11 (at `/usr/bin/python2`)
##### SUMMARY

Ansible fails when playbook contains `failed_when` with `not in [...]` with more than 1 element in list.
##### STEPS TO REPRODUCE

Run `ansible-playbook site.yml`

``` yaml

---

- hosts:
  - localhost
  tasks:
  - shell: /bin/false
    register: res
    failed_when: res.rc not in [0,2]
  - debug: msg={{ res }}
```
##### EXPECTED RESULTS

`ansible-playbook` finishes without a error.
##### ACTUAL RESULTS

`ansible-playbook` error:

```
TASK [command] *****************************************************************
fatal: [localhost]: FAILED! => {"failed": true, "msg": "The conditional check 'res.rc not in [1' failed. The error was: template error while templating string: unexpected '}', expected ']'. String: {% if res.rc not in [1 %} True {% else %} False {% endif %}"}
```

**Comment by sivel** [comment]
This is a known issue that has been fixed in devel already.

The work around is to use list syntax for failed_when such as:

```
failed_when:
  - res.rc not in [0,2]
```

**Comment by sivel** [comment]
Closing as a duplicate of https://github.com/ansible/ansible/issues/15291

**Comment by grossws** [comment]
@sivel, I hadn't found #15291. Sorry for inconvenience and thanks for workaround.

**Comment by sivel** [comment]
It's no problem.  The fix should be included in the upcoming 2.1 release.
